### ISSUE TITLE
panic: token map different size to token ring: got 1536 expected 2048

### ISSUE THREAD

**Issue by Fornax96** [scylladb/gocql#50]
Please answer these questions before submitting your issue. Thanks!

### What version of Scylla or Cassandra are you using?
Scylla 4.1

### What version of Gocql are you using?
github.com/scylladb/gocql v1.4.0

### What did you do?
I enabled TokenAwareHostPolicy like it says in the readme to see if performance would improve. On my test cluster it worked fine, but when I deployed it to production all my servers panicked with this error:
```
panic: token map different size to token ring: got 1536 expected 2048
goroutine 1 [running]:
github.com/gocql/gocql.(*networkTopology).replicaMap(0xc00051c088, 0xc000598380, 0xc00051c088, 0x0, 0x0)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/topology.go:273 +0x186d
github.com/gocql/gocql.(*tokenAwareHostPolicy).updateReplicas(0xc0001c2380, 0xc0003ca3a0, 0xc000023750, 0xa)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/policies.go:457 +0x25f
github.com/gocql/gocql.(*tokenAwareHostPolicy).KeyspaceChanged(0xc0001c2380, 0xc000023750, 0xa, 0x0, 0x0)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/policies.go:442 +0x99
github.com/gocql/gocql.(*Session).init(0xc000180700, 0xc0001a0380, 0x0)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/session.go:280 +0x5ca
github.com/gocql/gocql.NewSession(0xc00024f260, 0x3, 0x3, 0xb01191, 0x5, 0x0, 0x2540be400, 0x12a05f200, 0x23b6, 0xc000023750, ...)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/session.go:166 +0x809
github.com/gocql/gocql.(*ClusterConfig).CreateSession(...)
        /home/wim/Documents/workspace/go/pkg/mod/github.com/scylladb/gocql@v1.4.0/cluster.go:194
```

This actually brought the site down for a few minutes.

---

If you are having connectivity related issues please share the following additional information

### Describe your Cassandra cluster
please provide the following information

- output of `nodetool status`
```
Datacenter: eu-de
=================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address         Load       Tokens       Owns    Host ID                               Rack
UN  159.69.242.183  1.39 GB    256          ?       886551b1-7dbb-4ba3-8fe1-c388bd2fed53  rack1
UN  116.202.25.245  1.21 GB    256          ?       d8632268-2ec5-4f8a-bc33-5a20d8d8379d  rack1
UN  49.12.4.179     1.3 GB     256          ?       ce75f822-c8ec-4b81-9451-c6877b20ca3e  rack1
Datacenter: us-central
======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address         Load       Tokens       Owns    Host ID                               Rack
UN  209.126.0.59    1.44 GB    256          ?       b1b386a7-9997-4ba6-a79f-b24c9efa43c9  rack1
UN  207.244.245.11  1.17 GB    256          ?       a9027bea-5f82-4380-bede-892f882d9d17  rack1
Datacenter: us-west
===================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address         Load       Tokens       Owns    Host ID                               Rack
UN  23.175.1.55     1.53 GB    256          ?       ac84d564-01b3-4e04-9161-83b14cad1cca  rack1
UN  23.175.1.54     1.53 GB    256          ?       7b5b00fe-b7b6-4f0c-a896-5f59fef09bb2  rack1
UN  23.175.1.53     1.57 GB    256          ?       4b4bb1de-bb90-4dc9-b879-2b6f730a565e  rack1
```
- output of `SELECT peer, rpc_address FROM system.peers`
```

 peer           | rpc_address
----------------+----------------
    23.175.1.53 |    23.175.1.53
 159.69.242.183 | 159.69.242.183
    23.175.1.55 |    23.175.1.55
    23.175.1.54 |    23.175.1.54
   209.126.0.59 |   209.126.0.59
 207.244.245.11 | 207.244.245.11
 116.202.25.245 | 116.202.25.245
```
- rebuild your application with the `gocql_debug` tag and post the output
Is this necessary?

**Comment by mmatczuk** [comment]
From the numbers you have 8 nodes so the expectation 2048 is correct 1536 corresponds to 6 nodes. 
Something happened to us-central?

Does it reproduce?

The panic comes from `networkTopology`, does it reproduce with other settings?

> rebuild your application with the gocql_debug tag and post the output
> Is this necessary?

That would hopefully allow to see what gocql is doing.

**Comment by Fornax96** [comment]
I have removed the us-central nodes from the cluster as I wasn't really using them. Now it seems to work. Before removing the nodes I ran `nodetool repair -pr` on both and they seemed okay, so that's strange. There are some other nodes in the cluster where repair didn't finish and returned an error instead. Still trying to figure out what happened there.

I think it might be using the fallback method now because in Grafana the "Non-token aware queries" number is still the same.
